bench/BenchAll.hs

```diff
@@ -285,12 +336,80 @@ main = do
       , bgroup "Non-bounded encodings"
         [ benchB "byteStringHex"           byteStringData     $ byteStringHex
         , benchB "lazyByteStringHex"       lazyByteStringData $ lazyByteStringHex
-        , benchB "foldMap floatDec"        floatData          $ foldMap floatDec
-        , benchB "foldMap doubleDec"       doubleData         $ foldMap doubleDec
           -- Note that the small data corresponds to the intData pre-converted
           -- to Integer.
         , benchB "foldMap integerDec (small)"                     smallIntegerData        $ foldMap integerDec
         , benchB "foldMap integerDec (large)"                     largeIntegerData        $ foldMap integerDec
+        , bgroup "RealFloat"
+          [ bgroup "FGeneric"
+            [ bgroup "Positive"
+              [ benchB "Float"       floatPosData       $ foldMap (formatFloat  generic)
+              , benchB "Double"      doublePosData      $ foldMap (formatDouble generic)
+              , benchB "DoubleSmall" doublePosSmallData $ foldMap (formatDouble generic)
+              ]
+            , bgroup "Negative"
+              [ benchB "Float"       floatNegData  $ foldMap (formatFloat  generic)
+              , benchB "Double"      doubleNegData $ foldMap (formatDouble generic)
+              , benchB "DoubleSmall" doubleNegData $ foldMap (formatDouble generic)
+              ]
+            , bgroup "Special"
+              [ benchB "Float  Average" floatSpecials  $ foldMap (formatFloat  generic)
+              , benchB "Double Average" doubleSpecials $ foldMap (formatDouble generic)
+              ]
+            ]
+          , bgroup "FScientific"
+            [ bgroup "Positive"
+              [ benchB "Float"       floatPosData       $ foldMap (formatFloat  scientific)
+              , benchB "Double"      doublePosData      $ foldMap (formatDouble scientific)
+              , benchB "DoubleSmall" doublePosSmallData $ foldMap (formatDouble scientific)
+              ]
+            , bgroup "Negative"
+              [ benchB "Float"       floatNegData       $ foldMap (formatFloat  scientific)
+              , benchB "Double"      doubleNegData      $ foldMap (formatDouble scientific)
+              , benchB "DoubleSmall" doubleNegSmallData $ foldMap (formatDouble scientific)
+              ]
+            , bgroup "Special"
+              [ benchB "Float  Average" floatSpecials  $ foldMap (formatFloat  scientific)
+              , benchB "Double Average" doubleSpecials $ foldMap (formatDouble scientific)
+              ]
```

Comment:
This structure of several sorts of float inputs is repeated several times, please common it up with a function (INLINE if necessary).

Reply:
Seems like effort without much benefit that also makes it harder to add targeted benchmarks in the future.

Reply:
The effort I will expend asking myself "are these the same or is there a reason they are different?" the next time I read this code is comparable to the the effort it would take to de-duplicate it, so I think it's a reasonable investment if this code will be read more than once.

But both are less work than having this conversation, so it's not worth blocking you on if you'd rather not do it for whatever reason. We both have more important things to spend our time on!

Reply:
Much better, thank you!